### DISCUSSION TITLE
In mg_connection, what does the field data[] do exactly?

### DISCUSSION THREAD

**Discussion by humainbyte** [cesanta/mongoose#2653] [Ask questions, get answers] [answered]
In the pushing-live-data-on-http-connections (https://mongoose.ws/documentation/tutorials/webui-push/#pushing-live-data-on-http-connections), we have this chunk of code :

```
 else if (mg_http_match_uri(hm, "/api/log/live")) {
      c->data[0] = 'L';  // Mark that connection as live log listener
      mg_printf(c, "HTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\n\r\n");
    }
````
    
I am trying to find what the c->data[] is exactly. In the source code, it says this :
```
char data[MG_DATA_SIZE];     // Arbitrary connection data`
```
So it doesn't explain what data[] is in mg_connection and why putting c->data[0] = 'L' is used.

Is that explained somewhere? 

Thanks

**Comment by scaprile** [comment, marked as answer]
https://mongoose.ws/documentation/#struct-mg_connection

> Arbitrary connection data

Definitions from Oxford Languages: 
    1.    based on random choice or personal whim, rather than any reason or system. "an arbitrary decision"

Just a bunch of data so the user is able to put there whatever is needed. Can be used to mark connections in order to recognize specific ones when scanning through all. Can be used for whatever the user feels something can be stored in there.

https://github.com/cesanta/mongoose/blob/d7b2b3f42ec1109e6955baff6f836a15cfdd107a/examples/live-log/main.c#L17
https://github.com/cesanta/mongoose/blob/d7b2b3f42ec1109e6955baff6f836a15cfdd107a/examples/live-log/main.c#L34-L39
